contracts/evmscript/EVMScriptRegistry.sol

```diff
@@ -31,18 +33,32 @@ contract EVMScriptRegistry is IEVMScriptRegistry, EVMScriptRegistryConstants, Ar
         executors.push(ExecutorEntry(IEVMScriptExecutor(0), false));
     }
 
-    function addScriptExecutor(IEVMScriptExecutor _executor) external auth(REGISTRY_MANAGER_ROLE) returns (uint256 id) {
-        uint256 executorId = executors.push(ExecutorEntry(_executor, true));
+    function addScriptExecutor(IEVMScriptExecutor _executor) external auth(REGISTRY_ADD_EXECUTOR_ROLE) returns (uint256 id) {
+        uint256 executorId = executors.push(ExecutorEntry(_executor, true)) - 1;
```

Comment:
Note that this off-by-one was missed by everyone ;).

Reply:
Good catch!